broker/tasks/shield.py

```diff
@@ -0,0 +1,89 @@
+import logging
+
+from sqlalchemy.orm.attributes import flag_modified
+
+from broker.aws import shield
+from broker.extensions import db
+from broker.models import Operation
+from broker.tasks import huey
+
+logger = logging.getLogger(__name__)
+
+
+class ShieldProtections:
+    def __init__(self):
+        self.protected_cloudfront_ids: dict[str, str] = {}
+
+    def _list_cloudfront_protections(self):
+        paginator = shield.get_paginator("list_protections")
+        response_iterator = paginator.paginate(
+            InclusionFilters={"ResourceTypes": ["CLOUDFRONT_DISTRIBUTION"]},
+        )
+        for response in response_iterator:
+            for protection in response["Protections"]:
+                if "ResourceArn" in protection and "Id" in protection:
+                    self.protected_cloudfront_ids[protection["ResourceArn"]] = (
+                        protection["Id"]
+                    )
+
+    def get_cloudfront_protections(self):
+        if not self.protected_cloudfront_ids:
+            self._list_cloudfront_protections()
+        return self.protected_cloudfront_ids
+
+
+# Initialize here so that the class and its dictionary of mapped protections are
+# kept in memory for use by tasks
+shield_protections = ShieldProtections()
```

Comment:
I couldn't do this because it creates a circular dependency. `ShieldProtections` requires `shield` from `broker/aws.py` which itself imports `broker/extensions.py`